pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -202,9 +187,73 @@ func (c *Cluster) Start() error {
 	return nil
 }
 
+// ReconcileServices reconciles the services while requiring the active mgr to be detected
+func (c *Cluster) ReconcileServices(daemonNameToUpdate string) error {
+	return c.reconcileServices(daemonNameToUpdate, "")
+}
+
+// reconcile the services, if the active mgr is not detected, use the default mgr
+func (c *Cluster) reconcileServices(daemonNameToUpdate, defaultMgr string) error {
+	if daemonNameToUpdate != "" {
+		// If the services are already set to this daemon, no need to attempt to update
+		svc, err := c.context.Clientset.CoreV1().Services(c.clusterInfo.Namespace).Get(context.TODO(), AppName, metav1.GetOptions{})
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				logger.Warningf("mgr service %q not found, skipping the update", AppName)
+				return nil
+			}
+			logger.Errorf("failed to check current mgr service, proceeding to update. %v", err)
+		} else {
+			currentDaemon := svc.Spec.Selector[controller.DaemonIDLabel]
+			if currentDaemon == daemonNameToUpdate {
+				logger.Infof("mgr services already set to daemon %q, no need to update", daemonNameToUpdate)
+				return nil
+			}
+			logger.Infof("mgr service currently set to %q, checking if need to update to %q", currentDaemon, daemonNameToUpdate)
+		}
+	}
+
+	mgrMap, err := cephclient.CephMgrMap(c.context, c.clusterInfo)
+	if err != nil {
+		return errors.Wrap(err, "failed to get ceph status for the active mgr")
+	}
+	activeDaemon := mgrMap.ActiveName
+	if activeDaemon == "" {
+		if defaultMgr == "" {
+			return errors.New("active mgr not found")
+		}
+		logger.Infof("active mgr not found, defaulting to %q", defaultMgr)
```

Comment:
Would this be better as a warning?

Reply:
When the cluster is first reconciled and there are not yet any mgrs, this will be the normal case. But I'm thinking maybe the operator shouldn't own creating the services whether there are multiple mgrs. In that case, the sidecars could completely own creating and updating the services. I want to think about this a bit more...